library/src/webapp/editor/ckextraplugins/sakaipreview/plugin.js

```diff
@@ -103,22 +77,24 @@
 
         function getIframe(html) {
             var $iframe = $PBJQ('<iframe>');
-            $iframe.attr('src', 'data:text/html;charset=utf-8,' + encodeURI(html));
+            $iframe.attr('srcdoc', html);
             $iframe.attr('frameborder', '0');
             $iframe.attr('width', '100%');
             $iframe.attr('height', $PBJQ(window).height() - 240 + "px");
             return $iframe;
         }
 
+        var previewHighlightedOnly = false;
         function getEditorContent() {
-          var selectedText = "";
-          var selection = editor.getSelection();
+          //Available since CKEditor 4.5.0 to get selected Html. True to return html.
+          var selectedHtml = editor.getSelectedHtml(true)
```

Comment:
Codacy wanted a semi-colon

```suggestion
          var selectedHtml = editor.getSelectedHtml(true);
```